WinRT.Runtime/ComWrappersSupport.netstandard2.0.cs

```diff
@@ -75,8 +76,53 @@ public static T CreateRcwForComObject<T>(IntPtr ptr)
                 ABI.System.Nullable<Type> nt => (T)(object)nt.Value,
                 _ => (T)rcw
             };
-        }
-    
+        }
+
+        public static bool TryUnwrapObject(object o, out IObjectReference objRef)
+        {
+            // The unwrapping here needs to be in exact type match in case the user
+            // has implemented a WinRT interface or inherited from a WinRT class
+            // in a .NET (non-projected) type.
+
+            if (o is Delegate del)
+            {
+                return TryUnwrapObject(del.Target, out objRef);
+            }
+
+            Type type = o.GetType();
+
+            var objRefField = TypeObjectRefFieldCache.GetOrAdd(type, (type) =>
+            {
+                ObjectReferenceWrapperAttribute objRefWrapper = type.GetCustomAttribute<ObjectReferenceWrapperAttribute>();
+                if (objRefWrapper is object)
+                {
+                    return type.GetField(objRefWrapper.ObjectReferenceField, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+                }
+
+                ProjectedRuntimeClassAttribute projectedClass = type.GetCustomAttribute<ProjectedRuntimeClassAttribute>();
+                if (projectedClass is object && projectedClass.DefaultInterfaceProperty != null)
+                {
+                    return type.GetProperty(projectedClass.DefaultInterfaceProperty, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+                }
+
+                return null;
+            });
+
+
+            if (objRefField is FieldInfo field)

```

Comment:
you could avoid the type checks by keeping a cache of System.Funcs (factories) that take in o and return objRef

Reply:
I guess it was this way to start with - but would still help perf a bit

Reply:
Yea I already have the changes to address this feedback.  Working on fixing delegate marshaling.